app/controllers/home_pages_controller.rb

```diff
@@ -1,14 +1,33 @@
 class HomePagesController < ApplicationController
-  before_action :load_products, only: :index
+  before_action :load_product, only: :show
+
   def index
+    @products = Book.search(params[:search]).newest
     @pagy, @new_products = pagy @products, items: Settings.page_items
-    return @prices = @new_products.pluck(:price) unless params[:locale] == "vi"
+    return @prices = @new_products.pluck(:price) unless check_local
+
+    @prices = Book.dollar_to_vnds(@new_products)
+  end
+
+  def show
+    @book_id = @product.book_details.pluck(:id, :quantity).map do |a|
+      {id: a[0], quantity: a[1]}
+    end
+    return @price = @product.price unless check_local
 
-    @prices = Book.dollar_to_vnd(@new_products)
+    @price = @product.dollar_to_vnd @product
   end
 
   private
-  def load_products
-    @products = Book.search(params[:search]).newest
+  def load_product
+    @product = Book.includes(:book_details, :authors).find_by id: params[:id]
+    return if @product
+
+    flash[:danger] = t "book_not_found"
+    redirect_to root_path
+  end
+
+  def check_local
+    return true if params[:locale] == "vi"
   end
```

Comment:
```suggestion
  def vi_location?
    params[:locale] == "vi"
  end
```
Không hiểu thêm cái return làm gì